free-2.md

```diff
@@ -11,41 +11,88 @@ _What need is met by your idea? Who are you helping? What is that person's
+Currently, the common way of compiling and linking C/C++ code is through a DSL called
+make. However, make is very very confusing to use and doesnt seem to have much fluency
+so I think it really needs a revamp. If i could help them, their experience would be
+much less confusing. I imagine they could easily just define the dependencies in a more
+natural way than the Syntactic-Aspartame filled make. 
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+A DSL is required for this I feel. Currently lots of C/C++ rely on Make as a DSL
+and I would want my DSL to not be to big a change or too uncomfortable to use meaning
+a DSL is required to do this. I am sure a program could be created instead but it would
+take away from the current measures in place and could even make it more complicated.
+I think a DSL addresses the need since users will need to customize their files on the go
+and maybe even create these make files with scripts which is only possible if it is a DSL
+or has an API like a DSL.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+I think my main relation to this is that in CS70 make files were a horrible aspect
+and our groups could never figure them out. They just arent natural and I think my anger
+with the current system makes me very excited to fix it. 
+I came up with this when reading how a previous student recreated LaTex, another complicated
+DSL, and I thought about Make and wanted to maybe fix that too!
 
 ### Domain
 _Describe the project's domain in five words._
 
+Software development program compilation/linking
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+The user might interact with the language by specifying facts about the different
+C/C++ files they want to compile/link and then which compiler/tags they want to use.
+I am imagining a sort of text file where these facts are entered and then the language
+understands it and makes the correct calls. I think this is the right way to interact
+with the language since it is similar to how make files currently work but removes
+some of the more complicated aspects like having to understand exactly how to crate
+the CLI calls in the make file. 
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When a program runs, the different facts written down will be accumulated to determine 
+the structure of the code and then the proper compiler commands and tags will be used to
+compile/link everything. I think many errors could occur with not linking proper statements
+or not having all the edependiencies labeled and they would probably be communicated to 
```

Comment:
From my experience with the domain, we had a lot of trouble understanding the errors with the current version of make. It would be important to look at the current error messages make gives and see how or if those could be improved so that it clarifies where the error is coming from.